src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncRewriter.AsyncIteratorRewriter.cs

```diff
@@ -102,16 +105,14 @@ protected override void GenerateControlFields()
                     GeneratedNames.MakeAsyncIteratorPromiseIsActiveFieldName(), isPublic: true);
 
                 // Add a field: T current
-                _currentField = F.StateMachineField(method.IteratorElementType, GeneratedNames.MakeIteratorCurrentFieldName());
+                _currentField = F.StateMachineField(_elementType, GeneratedNames.MakeIteratorCurrentFieldName());
             }
 
             /// <summary>
             /// Generates the body of the replacement method, which initializes the state machine. Unlike regular async methods, we won't start it.
             /// </summary>
             protected override BoundStatement GenerateStateMachineCreation(LocalSymbol stateMachineVariable, NamedTypeSymbol frameType)
             {
-                // PROTOTYPE(async-streams): TODO review this (what is this error case at the start?)
```

Comment:
:memo: Removed this comment, as I don't think this case can be reached with async-iterators. But left the code as I don't think it's harmful.